### ISSUE TITLE
Is this project still active ?

### ISSUE THREAD

**Issue by silviucpp** [kr/beanstalkd#299]
Hello,

I see the activity on this project decreased a lot. Is still under active development ?

Silviu

**Comment by sergeyklay** [comment]
yes :) why not?

**Comment by sbrawner** [comment]
It is a mature, lean project that is stable and used in many enterprise production systems, including mine.

It’s not really a feature-rich project that requires lots of active development.

**Comment by silviucpp** [comment]
Hello,

I see a lot of pull requests for important fixes like : https://github.com/kr/beanstalkd/issues/205

And no reaction from maintainers. for this reason I asked this. 

Silviu

**Comment by ultrabug** [comment]
Let alone the project's activity, what's true is the lack of upstream answer and responsiveness which makes this project appear stale.

This indeed doesn't look good for an OSS and maybe the project would benefit from a more open list of contributors who could spare the time that @kr obviously have not (and it's totally normal & okay of course, our focus have to change over time).

> It is a mature, lean project that is stable and used in many enterprise production systems, including mine.

Same here, and we've proposed #295 but because of my points above, we have to use our own version in production which is a shame for a good project like beanstalkd imho.

I'd hate to see this project getting hurt by its stale state and contributors' frustration but can't do much about it I guess.

-- my 2 cents

**Comment by JensRantil** [comment]
@silviucpp I think your question has some truth to it. I've created #297 and #300 to improve maintainability of the project.

Let's wait a couple of days and see if @kr answers here. If not, I will send him a private e-mail asking if he's willing to invite more maintainers and possibly migrate this project to a separate Github organization. It was a while ago since I coded in C, but I'm willing to step up as a maintainer initially sorting through issues (closing dead ones), pull requests and fixing #297 & #300.

**Comment by silviucpp** [comment]
@JensRantil  what you propose sounds fair :) . At this moment I have no clue what's the most stable/active branch.
Silviu

**Comment by kr** [comment]
Hi folks. Yup as you can tell I haven't had time to devote to beanstalkd lately. This weekend (maybe Saturday, more likely Sunday) I will do a round of merging bug fixes from the various open PRs. Especially #295.

Ideally, if all we do is fix bugs, eventually beanstalkd will converge to something very very stable, and updates will be needed less and less often.

**Comment by ultrabug** [comment]
> Ideally, if all we do is fix bugs, eventually beanstalkd will converge to something very very stable, and updates will be needed less and less often.

This also depends on what you intend for the project mate, does that mean that any possible feature evolution won't get in any more ?

Cause I guess I can fairly say that heavy workload production users do need a bit more features (batching to name one) ;)

Thanks in advance for your time this weekend !

**Comment by JensRantil** [comment]
...and another question is, how can we make sure that the project doesn't stall and get quiet again?

**Comment by brailateo** [comment]
We are using beanstalk in production for more than 2 years and we need only in two occasions to restart it, probably due to the same problem of sparse hashmaps so we are glad to hear that it will be soon a new version available. It won't be probably ready in time to catch the Ubuntu 16.04 final release ...

**Comment by JensRantil** [comment]
I just wanted to chime in here that we all can help to keep this repository more alive by setting "Watching" on this repository so you get Github notifications on opened issues etc. Even if you are not familiar with the code, we can all help @kr by 1) answering questions and 2) making sure issues are well described (or recommend @kr to close it).

**Comment by silviucpp** [comment]
Hello 

I saw also @JensRantill commented on a lot of issues that they can be closed. Beside this I found also the following:

Inaccurate feedback 

https://github.com/kr/beanstalkd/issues/251

No feedback from reporter

https://github.com/kr/beanstalkd/issues/84
https://github.com/kr/beanstalkd/issues/245

Should be closed because were solved:

https://github.com/kr/beanstalkd/issues/99
https://github.com/kr/beanstalkd/issues/211
https://github.com/kr/beanstalkd/issues/212
https://github.com/kr/beanstalkd/issues/250
https://github.com/kr/beanstalkd/issues/197

Silviu

**Comment by kr** [comment]
Thanks so much @silviucpp! Much appreciated. :raised_hands:

**Comment by silviucpp** [comment]
Hello @kr ,

Another list of issues that can be closed is here ( @JensRantil  thanks a lot for commenting them): 

https://github.com/kr/beanstalkd/issues/314
https://github.com/kr/beanstalkd/issues/294
https://github.com/kr/beanstalkd/issues/268
https://github.com/kr/beanstalkd/issues/224 (I think it's already fixed on master but I'm not 100 % sure)
https://github.com/kr/beanstalkd/issues/140
https://github.com/kr/beanstalkd/issues/56
https://github.com/kr/beanstalkd/issues/256
https://github.com/kr/beanstalkd/issues/50
https://github.com/kr/beanstalkd/issues/27
https://github.com/kr/beanstalkd/issues/13 (Not sure 100 %)
https://github.com/kr/beanstalkd/issues/267
https://github.com/kr/beanstalkd/issues/261

Silviu

**Comment by silviucpp** [comment]
Also @kr 

I see this pull request https://github.com/kr/beanstalkd/pull/295 (the most important In my opinion) is still waiting for your feedback. Can you please review this in the next days to avoid becoming stale again.

Silviu

**Comment by kr** [comment]
> Can you please review this in the next days to avoid becoming stale again.

Please remember that I do all this work for free, in my spare time.

I try very hard to remember that everyone who shows up here is the same—a volunteer. And so I try to treat them with respect and gratitude. I only ask the same thing in return.

**Comment by kr** [comment]
Also, thanks both of you @JensRantil and @silviucpp for going through so many issues and updating them and helping out. It is making my life easier. :heart: 🙂

**Comment by silviucpp** [comment]
@kr now I think you can close also :

https://github.com/kr/beanstalkd/pull/295
https://github.com/kr/beanstalkd/issues/205

As time the fix was merged. 

Silviu

**Comment by silviucpp** [comment]
@kr can you make a release with all this changes that were merged and also generates new packages  ?
I think all this changes are pretty important and needs to go into a release.

Silviu

**Comment by acalfo** [comment]
+1

**Comment by gopi-ar** [comment]
@kr if you write instructions on creating a release version / distributing it to PPAs, we can fork and add a release..

**Comment by tomponline** [comment]
Any ideas when a 1.11 could be released? I think we are affected by from https://github.com/kr/beanstalkd/pull/295 and are considering build our RPM package from master as 1.11.1 

Thanks

**Comment by silviucpp** [comment]
Seems there is no activity from maintainers any longer. :( It's sad but true . In case anyone interested in deb packages I created a repo that allows you to generate them very fast: https://github.com/silviucpp/beanstalkd-deb

**Comment by stamster** [comment]
Sadly, beanstalkd won't get any attention after 3+ years w/o release.

**Comment by tomponline** [comment]
Its a shame, given that there is already stuff merged into master, it just needs tagging

**Comment by JensRantil** [comment]
I've e-mailed Keith and asked if he's willing to make me an maintainer.

**Comment by gopi-ar** [comment]
@kr / any of the future maintainers should consider a Patreon page for motivation. I know a bunch of companies that use beanstalkd that'll subscribe, us included.

**Comment by JensRantil** [comment]
I could make that happen if I become a maintainer.

**Comment by jellisii** [comment]
@JensRantil Perhaps time for a fork?

**Comment by pprindeville** [comment]
> Hi folks. Yup as you can tell I haven't had time to devote to beanstalkd lately. This weekend (maybe Saturday, more likely Sunday) I will do a round of merging bug fixes from the various open PRs. Especially #295.
> 
> Ideally, if all we do is fix bugs, eventually beanstalkd will converge to something very very stable, and updates will be needed less and less often.

Perhaps it's worth adding one or two more maintainers empowered to triage bugs, PR's, do merges of PR's, and even manage releases if @kr is too busy?

Looking at the project history, it seems like there are a few people with a long record of involvement.  @JensRantil @gnat42 @silviucpp?  Anyone want to volunteer?

**Comment by JensRantil** [comment]
Sorry guys that I haven't gotten back to you. I've had a fair amount of stuff to do both on the professional and private side of things. I received a response from Keith in an e-mail on May 15th this year. The relevant part is

> You’re right that I haven’t been able to find time for beanstalkd lately. If you want to take over maintenance, I’d suggest maybe you could fork the repo? If you like that idea, also free to let folks know about your fork in comments on issues here.
> 
> The beanstalk github org doesn’t really do much afaik.

So, I take that as we should fork the project (cc @jellisii). We need a name. I don't think "beanstalkd2" is a good name as there might a version two being released far in the future if it ever reawakens. Also, there's [AWS Elastic Beanstalk](https://aws.amazon.com/elasticbeanstalk/) and then there's also [BeanstalkApp](https://beanstalkapp.com) so it would be nice to get away from that name. "beand" is taken as a Github organization, also it might be a bit too close to Java-land :-) I couldn't find any good words related to "beans" (but I'm not a native English speaker). Some quick Googling on [English words ending with "*stalk"](https://www.wordfind.com/ends-with/stalk/) are "Leafstalk" and "Cornstalk" which are somewhat related to [Beanstalk]() (plant stalks).

Long story short...**cornstalkd** is now born. 🎉 You can find the Github organization [here]() and more specifically the repository [here](). I have also:

 * Enabled Travis CI integration for it.
 * Submitted a [pull request](https://github.com/cornstalkd/cornstalkd/pull/1) to Cornstalk to rename all references to Beanstalk.
 * Submitted a [pull request](https://github.com/kr/beanstalkd/pull/389) for this project to refer to the fork. Not sure @kr will merge it, but worth a shot.

I will now go through all open issues and pull requests and notify the relevant ones that there's a new (forked) kid in town.

Last, but not least, I am by no beans the best of C programmers, I don't have infinite amount of time to maintain the forked project (that said, I _do_ believe I'll be able to spend more time on beanstalkd...ehm, _cornstalkd_ 😉...than @kr) and I am not using beanstalkd in production on a day-to-day basis. I just think it's a very useful small application that has saved me a few times. What I'm trying to say is, I

1. want this to be a community effort. This means I'm open to invite other maintainers! @silviucpp, interested?
2. personally want the application to remain simple and keep its small scope. That said, that's also a community decision ;)

...and @stamster, it's never too late for a fork ;)

**Comment by pprindeville** [comment]
Well, first off, good job on being persevering and getting an answer back.  That's more than a lot of us dared hope for.

But I disagree with the assessment that the "beanstalk github org doesn’t really do much afaik".

Typically you can add others to the "team" and give them various rights, including the ability to do merges, close bugs, etc.

That seems like all you really need.

Doing a fork seems like more work than it's worth.

**Comment by JensRantil** [comment]
@pprindeville Oh, so you have write-access to do that? I don't. I'm not a maintainer of this repo. AFAIK, only Keith is.

**Comment by JensRantil** [comment]
...to clarify, I specifically asked if Keith was willing to give write access to other people to this repository in my e-mail to him. I offered to become a maintainer and move this project forward. I think his response was clear that he was unwilling to do that and instead recommended a fork.

**Comment by pprindeville** [comment]
Right. I'm saying the usual way of GitHub projects evolving, is when the sole maintainer runs out of steam, he delegates shared responsibility to others that he trusts.

I'm saying that it's a popular way to handle exactly this situation.

**Comment by JensRantil** [comment]
I understand. Tried that. Didn't work ;) Feel free to keep trying that if you believe that's still a viable way forward. Until then, I'll refer people to the fork.

**Comment by pprindeville** [comment]
> ...to clarify, I specifically asked if Keith was willing to give write access to other people to this repository in my e-mail to him. I offered to become a maintainer and move this project forward. I think his response was clear that he was unwilling to do that and instead recommended a fork.

I didn't see his entire response, but the part of it that I cited ("beanstalk github org doesn’t really do much afaik") suggests that he might not be aware of all of his options, and I was trying to point at least one of them out to him.

**Comment by JensRantil** [comment]
Ooops, noticed that my announcement had broken links to the new repo and organisation. Edited my post, but for clarity they are:

 * https://github.com/cornstalkd
 * https://github.com/cornstalkd/cornstalkd

**Comment by JensRantil** [comment]
My original e-mail to Keith:

> Hey Keith,
> 
> I'll try to keep this short as I'm sure you are a busy man:
> 
> My name is Jens and I'm a Swedish software dev who's been keeping an eye on beanstalkd over the years. There's a long-standing issue about the project being dead and I was wondering if you're interested in handing over the torch to another maintainer? If so, I offer to step up and 1) try to do some low-energy maintenance or 2) long-term find an even more active maintainer going forward.
> 
> I understand, and respect, your quest for minimalism and I'm sure that the project can remain rooted in simplicity. The project could even be considered feature complete if that's what you want, but it needs bug fixes merged, someone to answer issues and who can make releases.
> 
> If you feel like this sounds like a good way forward, please make me an admin for the repo. My Github username is "JensRantil".
> 
> Also, I noticed your Github user is part of the beanstalkd organization. Would it make sense to migrate the beanstalkd repo to that organization?
> 
> Cheers,
> Jens

**Comment by pprindeville** [comment]
One of the issues with a fork is that there's no easy way to migrate PR's to it, without asking the original author to do even more effort for possibly no return.

**Comment by JensRantil** [comment]
Absolutely. I understand, I hear ya, but I'm afraid there's nothing I can do about that. Keith is a busy man with other priorities. We need to move the beanstalkd to another place.

**Comment by tomponline** [comment]
@JensRantil this is great, thanks for taking the initiative on this one.

**Comment by silviucpp** [comment]
@pprindeville I think it's possible to move the PR's without any problem. Just add as another remote branch the one from the PR and then you can merge it in your fork.

**Comment by JensRantil** [comment]
Hello again,

Keith got back to me as I announced the fork to him last night. He made me an owner of the beanstalkd GitHub organization. I will go that route instead. But before I do that I am awaiting response from Keith if he's willing to transfer this repository to the organization (to preserve pull requests and issues) and optionally create a fork on his user. If he's not, I will create a new repo and push Keith's repo's master there.

**Comment by JensRantil** [comment]
Hello again. Good news! Keith agreed to transfer beanstalkd to the beanstalkd organization! As you can see, he did the transfer some time these past 8 hours. This means that all pull requests and issues don't need to be resubmitted _and_ we have the access we need to build an active community around this.

I have deleted the cornstalkd fork and its organization and I'm happy we were able to avoid a fork!

**Comment by sergeyklay** [comment]
@JensRantil Could you or someone who has the appropriate authority create a repository in beanstalkd org for php extension implementation (C-89) so I can push my private project.

**Comment by ysmolsky** [comment]
No activity from PR approvers in last year. Should reopen this issue?
